### DISCUSSION TITLE
Check failed: index < Size() (XX vs. XX)

### DISCUSSION THREAD

**Discussion by FreddyHunter** [google/or-tools#2705] [Routing (and legacy CP) questions]
Dear Guys,

I am running a CVRPTW in python with different starts and ends, but I am getting the following error message when I run the model.

```
F0330 08:38:27.395263 25730 routing.cc:1630] Check failed: index < Size() (10 vs. 10)
*** Check failure stack trace: ***
Aborted
```
My distance Matrix is:
```python
    data['distance_matrix'] = [
[0, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000],
[1000, 0, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000],
[2000, 1000, 0, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000],
[3000, 2000, 1000, 0, 4000, 5000, 6000, 7000, 8000, 9000, 10000],
[4000, 3000, 2000, 1000, 0, 5000, 6000, 7000, 8000, 9000, 10000],
[5000, 4000, 3000, 2000, 1000, 0, 6000, 7000, 8000, 9000, 10000],
[6000, 5000, 4000, 3000, 2000, 1000, 0, 7000, 8000, 9000, 10000],
[7000, 6000, 5000, 4000, 3000, 2000, 1000, 0, 8000, 9000, 10000],
[8000, 7000, 6000, 5000, 4000, 3000, 2000, 1000, 0, 9000, 10000],
[9000, 8000, 7000, 6000, 5000, 4000, 3000, 2000, 1000, 0, 10000],
[10000, 9000, 8000, 7000, 6000, 5000, 4000, 3000, 2000, 1000, 0] 
```
My starts and ends vectors are:
```py
    data['starts'] = [0, 0, 0, 4, 5]
    data['ends'] = [0, 0, 1, 7, 8]
```
If I change the vectors to:
```py
    data['starts'] = [0, 0, 0, 4, 5]
    data['ends'] = [0, 0, 0, 7, 8]
```
it run properly respecting the constraints and giving me a solution.

If I also change the vectors to:
```py
    data['starts'] = [0, 0, 3, 4, 5]
    data['ends'] = [0, 0, 1, 7, 8]
```
I get the same error:
```
F0403 11:35:27.459228 25743 routing.cc:1630] Check failed: index < Size() (9 vs. 9)
*** Check failure stack trace: ***
Aborted
```

If I change it to:
```py
    data['starts'] = [1, 2, 3, 4, 5]
    data['ends'] = [6, 7, 8, 9, 10]
```
I get the same error:
```
F0404 22:58:40.714843 25752 routing.cc:1630] Check failed: index < Size() (6 vs. 6)
*** Check failure stack trace: ***
Aborted
```

So my doubts are:
1) I would like to understand what is causing this behavior, and why with three different starts and ends it run properly. And why it does not run when we include at least one more start or end.

2) If I have different warehouses (starts) with different garages (ends) is there a rule how to mount the matrices? I mean, first positions need to be for starts, then delivery points and finally the end points, or it does not matter for the optimizer?

Thanks for comments, tips, suggestions and explanations.

**Comment by Mizux** [comment]
Did you correctly convert indices in your transit callback ?

Please note you **can't** use `manager.NodeToIndex()` for start and end nodes, you must use `routing.Start(vehicle_id)` and `routing.End(vehicle_id)` respectively
